.github/workflows/maven-verify.yml

```diff
@@ -208,7 +208,12 @@ jobs:
         shell: bash
 
       - name: Clean Ensuring no file handle remains open on windows
-        run: ./mvnw clean --errors --batch-mode --show-version
+        run: ./mvnw clean --errors --batch-mode --show-version   
+        
+      - name: clean local repo to ensure no locks remains on files
+        if: always()
+        run: rm -rf ~/.m2/
```

Comment:
.m2 is cached by action .... when we remove content cache will be empty and next build will download everything again

Reply:
so we cannot test if our build/library/plugins are locking files?

Reply:
I have never been really sure if this cache works or how this cache works.... 
but looking at builds everywhere (this one as an example just look at this one https://github.com/apache/maven-compiler-plugin/runs/7168529515?check_suite_focus=true ) there are always tons of `[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/` for artifacts which are supposed to been already cached.
Not really sure how it is efficient. I have always sceptical about this maybe there are some bad configuration?

look at the debug I added some debug and `~/.m2/repository` is empty after the setup java which is supposed to handle the cache.
Interesting I will try to have a look at the source code of this action/cache.

Reply:
cache is removed when is not used for 7 days - compiler-p case - look into `Set up JDK` and  `Post Set up JDK`

case in this test - first build did `rm ..` cache was updated with empty .m2 and next build has cache by id - but it is empty.

for test you can change someting in pom.xml so we will new cache id - cache id for maven build is computed based on poms like `${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}`

Reply:
but here in this PR I did few more builds after removing the `rm` and didn't touch any pom.xml.
cache still empty as shown by `ls -alrt ~/.m2/`
if every single change in pom.xml clear the cache and 7 days, not sure it is very efficient really :)